core-server/src/main/java/org/glassfish/jersey/server/internal/RuntimeDelegateImpl.java

```diff
@@ -39,11 +60,126 @@ public RuntimeDelegateImpl() {
+        return new JAXRS.Configuration.Builder() {
+            private final Map<String, Object> properties = new HashMap<>();
+
+            {
+                this.properties.put(JAXRS.Configuration.PROTOCOL, "HTTP");
+                this.properties.put(JAXRS.Configuration.HOST, "localhost");
+                this.properties.put(JAXRS.Configuration.PORT, -1); // Auto-select port 80 for HTTP or 443 for HTTPS
+                this.properties.put(JAXRS.Configuration.ROOT_PATH, "/");
+                this.properties.put(ServerProperties.HTTP_SERVER_CLASS, Server.class); // Auto-select first provider
+                try {
+                    this.properties.put(JAXRS.Configuration.SSL_CONTEXT, SSLContext.getDefault());
+                } catch (final NoSuchAlgorithmException e) {
+                    throw new RuntimeException(e);
+                }
+                this.properties.put(JAXRS.Configuration.SSL_CLIENT_AUTHENTICATION,
+                        JAXRS.Configuration.SSLClientAuthentication.NONE);
+                this.properties.put(ServerProperties.AUTO_START, TRUE);
+            }
+
+            @Override
+            public final JAXRS.Configuration.Builder property(final String name, final Object value) {
+                this.properties.put(name, value);
+                return this;
+            }
+
+            @SuppressWarnings("unchecked")
+            @Override
+            public final <T> Builder from(final BiFunction<String, Class<T>, Optional<T>> configProvider) {
+                PROPERTY_TYPES.forEach(
+                        (propertyName, propertyType) -> configProvider.apply(propertyName, (Class<T>) propertyType)
+                                .ifPresent(propertyValue -> this.properties.put(propertyName, propertyValue)));
+                return this;
+            }
+
+            @Override
+            public final Builder from(final Object externalConfig) {
+                if (externalConfig instanceof Config) {
+                    return this.from((Config) externalConfig);
+                }
+
+                return this;
+            }
+
+            private final Builder from(final Config config) {
+                return this.from(config::getOptionalValue);
+            }
+
+            @Override
+            public final JAXRS.Configuration build() {
+                return this.properties::get;
+            }
+        };
+    }
+
+    @SuppressWarnings("unchecked")
+    @Override
+    public CompletableFuture<JAXRS.Instance> bootstrap(final Application application,
+            final JAXRS.Configuration configuration) {
+        return CompletableFuture.supplyAsync(() -> {
+            final Class<Server> httpServerClass = (Class<Server>) configuration
+                    .property(ServerProperties.HTTP_SERVER_CLASS);
+
+            return new JAXRS.Instance() {
+                private final Server server = ServerFactory.createServer(httpServerClass, application, configuration);
```

Comment:
usually, `configuration.property(ServerProperties.HTTP_SERVER_CLASS)` won't be set. so httpServerClass will be `null` and  ServerFactory.createServer will cast to httpServerClass==null

Reply:
This won't happen, because the property has a default set *by the config builder*, see:
https://github.com/eclipse-ee4j/jersey/blob/fa9d0fe2d39be5e475035ac626a1bf19e0290753/core-server/src/main/java/org/glassfish/jersey/server/internal/RuntimeDelegateImpl.java#L94